api/client/build.go

```diff
@@ -575,6 +577,28 @@ func rewriteDockerfileFrom(dockerfileName string, translator func(string, regist
 		}
 	}()
 
+	var signals []os.Signal
+	for _, sig := range []string{"INT", "TERM", "QUIT"} {
+		s, ok := signal.SignalMap[sig]
+		if ok {
+			signals = append(signals, s)
+		}
+	}
+	if len(signals) > 0 {
+		sigc := make(chan os.Signal, 1)
+		gosignal.Notify(sigc, signals...)
```

Comment:
@crosbymichael  The tempfile suppose  be closed at https://github.com/docker/docker/blob/master/api/client/build.go#L125 by the caller if the error is nil.  so we can't close it at https://github.com/docker/docker/pull/17060#discussion-diff-42164667L570 if error is not nil